README.md
# OpenCloset-Common
Common perl libs of OpenCloset


Changes
v0.1.12     2021-05-17 04:55:06 Asia/Seoul
  - 착용여부 목록에서 "입고감+기록요망"을 "보관요망"으로 변경 (#7)

v0.1.11   Sat, 12 Dec 2020 10:38:37 +0900
  - iamport 결제수단 업데이트

v0.1.10   Sat, 11 May 2019 14:08:31 +0900
  - '입고감' 카테고리에 '온라인예정' 추가

v0.1.9    Sun, 16 Sep 2018 13:55:24 +0900
  - Add $MONITOR_TTS_TO_INDEX and $MONITOR_TTS_TO_ROOM

v0.1.8    Tue, 03 Jul 2018 16:21:12 +0900
  - Add $MAX_EXTENSION_DAYS

v0.1.7    Sat, 24 Mar 2018 12:21:32 +0900
  - 주문서가 미납인지 판별하는 `OpenCloset::Common::Unpaid::is_unpaid` 함수 추가 (#5)

v0.1.6    Wed, 21 Mar 2018 23:45:26 +0900
  - 주문서에 '입고감'과 관련된 상수 추가

v0.1.5    Thu, 23 Nov 2017 11:23:11 +0900
  - Add %MAX_SUIT_TYPE_COUPON_PRICE
  - Remove $MAX_SUIT_COUPON_PRICE

v0.1.4    Tue, 21 Nov 2017 16:14:21 +0900
  - Add $MAX_SUIT_COUPON_PRICE

v0.1.3    Wed, 15 Nov 2017 14:49:24 +0900
  - Add SHIPPING_DEADLINE_HOUR constants

v0.1.2    Tue, 24 Oct 2017 15:23:27 +0900
  - Add missing deps

v0.1.1    Thu, 04 May 2017 15:58:54 +0900
  - "재활용(비전)" 과 관련된 상태를 추가

v0.1.0    Mon, 01 May 2017 16:36:30 +0900
  - 반납 -> 환불 기능 추가 (#4)

v0.0.17   Tue, 18 Apr 2017 16:02:34 +0900
  - Add OpenCloset::Common::Unpaid::merchant_uid
  - Add OpenCloset::Common::Unpaid::create_vbank

v0.0.16   Mon, 10 Apr 2017 13:08:02 +0900
  - Add %OpenCloset::Constants::Measurement::AVG_LEG_BY_HEIGHT
  - Add %OpenCloset::Constants::Measurement::AVG_KNEE_BY_HEIGHT

v0.0.15   Mon, 20 Mar 2017 17:02:51 +0900
  - OpenCloset::Common::Unpaid 모듈 추가 (#3)

v0.0.14   Tue, 07 Mar 2017 15:14:55 +0900
  - Category 에 BAG 을 추가

v0.0.13   Wed, 15 Feb 2017 12:40:08 +0900
  - Remove $RENTAL_BUFFER and $PARCEL_BUFFER
  - Fix $SHIPPING_BUFFER from 3 to 2

v0.0.12   Tue, 14 Feb 2017 12:27:47 +0900
  - Add %PAY_METHOD_MAP

v0.0.11   Fri, 10 Feb 2017 13:21:02 +0900
  - Add OpenCloset::Constants

v0.0.10   Thu, 19 Jan 2017 12:26:46 +0900
  - export $WAITING_DEPOSIT

v0.0.9    Thu, 19 Jan 2017 12:09:11 +0900
  - 입금대기 상태를 추가

v0.0.8    Tue, 03 Jan 2017 13:21:58 +0900
  - 온라인 상태를 update
  - @OpenCloset::Constants::Measurement::PRIMARY 에 윗배와 엉덩이 둘레를 추가

v0.0.7    Fri, 01 Jul 2016 15:17:57 +0900
  - 사이즈제안에서 여성팬츠를 고려

v0.0.6    Fri, 20 May 2016 17:42:37 +0900
  - category 의 export 변수 추가

v0.0.5    Fri, 20 May 2016 17:17:17 +0900
  - Export category hash MAPs

v0.0.4    Sat, 14 May 2016 05:45:57 +0900
  - resize 된 허리 둘레에 맞춰서 밑단을 계산하도록 함

v0.0.3    Thu, 12 May 2016 13:12:24 +0900
  - Fix assigned clothes top, bottom syntax

v0.0.2    Thu, 12 May 2016 12:34:40 +0900
  - 정해진 표를 통해 의류의 수선 사이즈를 제안
  - Add OpenCloset::Constants::Status
  - Add OpenCloset::Common::Clothes

v0.0.1    Mon, 02 May 2016 18:56:40 +0900
  - Initial release


lib/OpenCloset/Constants/Status.pm
package OpenCloset::Constants::Status;

use utf8;
require Exporter;
@ISA       = qw/Exporter/;
@EXPORT_OK = qw/$RENTABLE $RENTAL $RENTALESS $RESERVATION $CLEANING $REPAIR $LOST $DISCARD $RETURNED $PARTIAL_RETURNED $RETURNING
    $NOT_VISITED $VISITED $RESERVATED $RETURN_RESERVATED $MEASUREMENT $SELECT $BOX $PAYMENT $FITTING_ROOM1 $FITTING_ROOM2
    $FITTING_ROOM3  $FITTING_ROOM4  $FITTING_ROOM5  $FITTING_ROOM6  $FITTING_ROOM7  $FITTING_ROOM8  $FITTING_ROOM9  $FITTING_ROOM10
    $FITTING_ROOM11 $FITTING_ROOM12 $FITTING_ROOM13 $FITTING_ROOM14 $FITTING_ROOM15 $FITTING_ROOM16 $FITTING_ROOM17 $FITTING_ROOM18
    $FITTING_ROOM19 $FITTING_ROOM20 $NOT_RENTAL $CANCEL_BOX $PAYBACK $NO_SIZE $BOXED $RECYCLE_1 $RECYCLE_2 $UNRECYCLE $CHOOSE_CLOTHES $CHOOSE_ADDRESS $PAYMENT_DONE $DEPOSIT_CONFIRMED $WAITING_SHIPPED $SHIPPED $DELIVERED $RETURN_REQUESTED $WAITING_DEPOSIT $RECYCLE_3/;

our $RENTABLE          = 1;
our $RENTAL            = 2;
our $RENTALESS         = 3;  # FIXME
our $RESERVATION       = 4;
our $CLEANING          = 5;
our $REPAIR            = 6;
our $LOST              = 7;
our $DISCARD           = 8;
our $RETURNED          = 9;
our $PARTIAL_RETURNED  = 10;
our $RETURNING         = 11;
our $NOT_VISITED       = 12;
our $VISITED           = 13;
our $RESERVATED        = 14;
our $RETURN_RESERVATED = 15;
our $MEASUREMENT       = 16;
our $SELECT            = 17;
our $BOX               = 18;
our $PAYMENT           = 19;
our $FITTING_ROOM1     = 20;
our $FITTING_ROOM2     = 21;
our $FITTING_ROOM3     = 22;
our $FITTING_ROOM4     = 23;
our $FITTING_ROOM5     = 24;
our $FITTING_ROOM6     = 25;
our $FITTING_ROOM7     = 26;
our $FITTING_ROOM8     = 27;
our $FITTING_ROOM9     = 28;
our $FITTING_ROOM10    = 29;
our $FITTING_ROOM11    = 30;
our $FITTING_ROOM12    = 31;
our $FITTING_ROOM13    = 32;
our $FITTING_ROOM14    = 33;
our $FITTING_ROOM15    = 34;
our $FITTING_ROOM16    = 35;
our $FITTING_ROOM17    = 36;
our $FITTING_ROOM18    = 37;
our $FITTING_ROOM19    = 38;
our $FITTING_ROOM20    = 39;
our $NOT_RENTAL        = 40;
our $CANCEL_BOX        = 41;
our $PAYBACK           = 42;
our $NO_SIZE           = 43;
our $BOXED             = 44;
our $RECYCLE_1         = 45;
our $RECYCLE_2         = 46;
our $UNRECYCLE         = 47; # FIXME
our $CHOOSE_CLOTHES    = 48;
our $CHOOSE_ADDRESS    = 49;
our $PAYMENT_DONE      = 50;
our $DEPOSIT_CONFIRMED = 51;
our $WAITING_SHIPPED   = 52;
our $SHIPPED           = 53;
our $DELIVERED         = 54;
our $RETURN_REQUESTED  = 55;
our $WAITING_DEPOSIT   = 56;
our $RECYCLE_3         = 57;

our $LABEL_RENTABLE          = '대여가능';
our $LABEL_RENTAL            = '대여중';
our $LABEL_RENTALESS         = '대여불가';     # FIXME
our $LABEL_RESERVATION       = '예약';
our $LABEL_CLEANING          = '세탁';
our $LABEL_REPAIR            = '수선';
our $LABEL_LOST              = '분실';
our $LABEL_DISCARD           = '폐기';
our $LABEL_RETURNED          = '반납';
our $LABEL_PARTIAL_RETURNED  = '부분반납';
our $LABEL_RETURNING         = '반납배송중';
our $LABEL_NOT_VISITED       = '방문안함';
our $LABEL_VISITED           = '방문';
our $LABEL_RESERVATED        = '방문예약';
our $LABEL_RETURN_RESERVATED = '배송예약';
our $LABEL_MEASUREMENT       = '치수측정';
our $LABEL_SELECT            = '의류준비';
our $LABEL_BOX               = '포장';
our $LABEL_PAYMENT           = '결제대기';
our $LABEL_FITTING_ROOM1     = '탈의01';
our $LABEL_FITTING_ROOM2     = '탈의02';
our $LABEL_FITTING_ROOM3     = '탈의03';
our $LABEL_FITTING_ROOM4     = '탈의04';
our $LABEL_FITTING_ROOM5     = '탈의05';
our $LABEL_FITTING_ROOM6     = '탈의06';
our $LABEL_FITTING_ROOM7     = '탈의07';
our $LABEL_FITTING_ROOM8     = '탈의08';
our $LABEL_FITTING_ROOM9     = '탈의09';
our $LABEL_FITTING_ROOM10    = '탈의10';
our $LABEL_FITTING_ROOM11    = '탈의11';
our $LABEL_FITTING_ROOM12    = '탈의12';
our $LABEL_FITTING_ROOM13    = '탈의13';
our $LABEL_FITTING_ROOM14    = '탈의14';
our $LABEL_FITTING_ROOM15    = '탈의15';
our $LABEL_FITTING_ROOM16    = '탈의16';
our $LABEL_FITTING_ROOM17    = '탈의17';
our $LABEL_FITTING_ROOM18    = '탈의18';
our $LABEL_FITTING_ROOM19    = '탈의19';
our $LABEL_FITTING_ROOM20    = '탈의20';
our $LABEL_NOT_RENTAL        = '대여안함';
our $LABEL_CANCEL_BOX        = '포장취소';
our $LABEL_PAYBACK           = '환불';
our $LABEL_NO_SIZE           = '사이즈없음';
our $LABEL_BOXED             = '포장완료';
our $LABEL_RECYCLE_1         = '재활용(옷캔)';
our $LABEL_RECYCLE_2         = '재활용(비백)';
our $LABEL_UNRECYCLE         = '사용못함';     # FIXME
our $LABEL_CHOOSE_CLOTHES    = '의류선택';
our $LABEL_CHOOSE_ADDRESS    = '주소선택';
our $LABEL_PAYMENT_DONE      = '결제완료';
our $LABEL_DEPOSIT_CONFIRMED = '입금확인';
our $LABEL_WAITING_SHIPPED   = '발송대기';
our $LABEL_SHIPPED           = '배송중';
our $LABEL_DELIVERED         = '배송완료';
our $LABEL_RETURN_REQUESTED  = '반송신청';
our $LABEL_WAITING_DEPOSIT   = '입금대기';
our $LABEL_RECYCLE_3         = '재활용(비전)';

our @ALL = (
    $RENTABLE, $RENTAL, $RENTALESS, $RESERVATION, $CLEANING, $REPAIR, $LOST, $DISCARD, $RETURNED, $PARTIAL_RETURNED, $RETURNING,
    $NOT_VISITED, $VISITED, $RESERVATED, $RETURN_RESERVATED, $MEASUREMENT, $SELECT, $BOX, $PAYMENT, $FITTING_ROOM1, $FITTING_ROOM2,
    $FITTING_ROOM3,  $FITTING_ROOM4,  $FITTING_ROOM5,  $FITTING_ROOM6,  $FITTING_ROOM7,  $FITTING_ROOM8,  $FITTING_ROOM9,  $FITTING_ROOM10,
    $FITTING_ROOM11, $FITTING_ROOM12, $FITTING_ROOM13, $FITTING_ROOM14, $FITTING_ROOM15, $FITTING_ROOM16, $FITTING_ROOM17, $FITTING_ROOM18,
    $FITTING_ROOM19, $FITTING_ROOM20, $NOT_RENTAL, $CANCEL_BOX, $PAYBACK, $NO_SIZE, $BOXED, $RECYCLE_1, $RECYCLE_2, $UNRECYCLE,
    $CHOOSE_CLOTHES,  $CHOOSE_ADDRESS, $PAYMENT_DONE, $DEPOSIT_CONFIRMED, $WAITING_SHIPPED, $SHIPPED, $DELIVERED, $RETURN_REQUESTED,
    $WAITING_DEPOSIT, $RECYCLE_3
);

our @ALL_LABEL = (
    $LABEL_RENTABLE, $LABEL_RENTAL,           $LABEL_RENTALESS, $LABEL_RESERVATION, $LABEL_CLEANING, $LABEL_REPAIR, $LABEL_LOST, $LABEL_DISCARD,
    $LABEL_RETURNED, $LABEL_PARTIAL_RETURNED, $LABEL_RETURNING,
    $LABEL_NOT_VISITED, $LABEL_VISITED,       $LABEL_RESERVATED, $LABEL_RETURN_RESERVATED, $LABEL_MEASUREMENT, $LABEL_SELECT, $LABEL_BOX,
    $LABEL_PAYMENT,     $LABEL_FITTING_ROOM1, $LABEL_FITTING_ROOM2,
    $LABEL_FITTING_ROOM3,  $LABEL_FITTING_ROOM4,  $LABEL_FITTING_ROOM5,  $LABEL_FITTING_ROOM6,  $LABEL_FITTING_ROOM7,  $LABEL_FITTING_ROOM8,
    $LABEL_FITTING_ROOM9,  $LABEL_FITTING_ROOM10,
    $LABEL_FITTING_ROOM11, $LABEL_FITTING_ROOM12, $LABEL_FITTING_ROOM13, $LABEL_FITTING_ROOM14, $LABEL_FITTING_ROOM15, $LABEL_FITTING_ROOM16,
    $LABEL_FITTING_ROOM17, $LABEL_FITTING_ROOM18,
    $LABEL_FITTING_ROOM19, $LABEL_FITTING_ROOM20, $LABEL_NOT_RENTAL, $LABEL_CANCEL_BOX, $LABEL_PAYBACK, $LABEL_NO_SIZE, $LABEL_BOXED,
    $LABEL_RECYCLE_1,      $LABEL_RECYCLE_2,      $LABEL_UNRECYCLE,
    $LABEL_CHOOSE_CLOTHES, $LABEL_CHOOSE_ADDRESS, $LABEL_PAYMENT_DONE, $LABEL_DEPOSIT_CONFIRMED, $LABEL_WAITING_SHIPPED, $LABEL_SHIPPED,
    $LABEL_DELIVERED,
    $LABEL_RETURN_REQUESTED, $LABEL_WAITING_DEPOSIT, $LABEL_RECYCLE_3
);

our %LABEL_MAP = (
    $RENTABLE          => $LABEL_RENTABLE,
    $RENTAL            => $LABEL_RENTAL,
    $RENTALESS         => $LABEL_RENTALESS,
    $RESERVATION       => $LABEL_RESERVATION,
    $CLEANING          => $LABEL_CLEANING,
    $REPAIR            => $LABEL_REPAIR,
    $LOST              => $LABEL_LOST,
    $DISCARD           => $LABEL_DISCARD,
    $RETURNED          => $LABEL_RETURNED,
    $PARTIAL_RETURNED  => $LABEL_PARTIAL_RETURNED,
    $RETURNING         => $LABEL_RETURNING,
    $NOT_VISITED       => $LABEL_NOT_VISITED,
    $VISITED           => $LABEL_VISITED,
    $RESERVATED        => $LABEL_RESERVATED,
    $RETURN_RESERVATED => $LABEL_RETURN_RESERVATED,
    $MEASUREMENT       => $LABEL_MEASUREMENT,
    $SELECT            => $LABEL_SELECT,
    $BOX               => $LABEL_BOX,
    $PAYMENT           => $LABEL_PAYMENT,
    $FITTING_ROOM1     => $LABEL_FITTING_ROOM1,
    $FITTING_ROOM2     => $LABEL_FITTING_ROOM2,
    $FITTING_ROOM3     => $LABEL_FITTING_ROOM3,
    $FITTING_ROOM4     => $LABEL_FITTING_ROOM4,
    $FITTING_ROOM5     => $LABEL_FITTING_ROOM5,
    $FITTING_ROOM6     => $LABEL_FITTING_ROOM6,
    $FITTING_ROOM7     => $LABEL_FITTING_ROOM7,
    $FITTING_ROOM8     => $LABEL_FITTING_ROOM8,
    $FITTING_ROOM9     => $LABEL_FITTING_ROOM9,
    $FITTING_ROOM10    => $LABEL_FITTING_ROOM10,
    $FITTING_ROOM11    => $LABEL_FITTING_ROOM11,
    $FITTING_ROOM12    => $LABEL_FITTING_ROOM12,
    $FITTING_ROOM13    => $LABEL_FITTING_ROOM13,
    $FITTING_ROOM14    => $LABEL_FITTING_ROOM14,
    $FITTING_ROOM15    => $LABEL_FITTING_ROOM15,
    $FITTING_ROOM16    => $LABEL_FITTING_ROOM16,
    $FITTING_ROOM17    => $LABEL_FITTING_ROOM17,
    $FITTING_ROOM18    => $LABEL_FITTING_ROOM18,
    $FITTING_ROOM19    => $LABEL_FITTING_ROOM19,
    $FITTING_ROOM20    => $LABEL_FITTING_ROOM20,
    $NOT_RENTAL        => $LABEL_NOT_RENTAL,
    $CANCEL_BOX        => $LABEL_CANCEL_BOX,
    $PAYBACK           => $LABEL_PAYBACK,
    $NO_SIZE           => $LABEL_NO_SIZE,
    $BOXED             => $LABEL_BOXED,
    $RECYCLE_1         => $LABEL_RECYCLE_1,
    $RECYCLE_2         => $LABEL_RECYCLE_2,
    $UNRECYCLE         => $LABEL_UNRECYCLE,
    $CHOOSE_CLOTHES    => $LABEL_CHOOSE_CLOTHES,
    $CHOOSE_ADDRESS    => $LABEL_CHOOSE_ADDRESS,
    $PAYMENT_DONE      => $LABEL_PAYMENT_DONE,
    $DEPOSIT_CONFIRMED => $LABEL_DEPOSIT_CONFIRMED,
    $WAITING_SHIPPED   => $LABEL_WAITING_SHIPPED,
    $SHIPPED           => $LABEL_SHIPPED,
    $DELIVERED         => $LABEL_DELIVERED,
    $RETURN_REQUESTED  => $LABEL_RETURN_REQUESTED,
    $WAITING_DEPOSIT   => $LABEL_WAITING_DEPOSIT,
    $RECYCLE_3         => $LABEL_RECYCLE_3,
);

our %COLOR_MAP = (
    $RENTABLE        => 'label-success',
    $RENTAL          => 'label-danger',
    $RENTALESS       => 'label-danger',
    $LOST            => 'label-danger',
    $DISCARD         => 'label-danger',
    $WAITING_SHIPPED => 'label-success',
);

map { $COLOR_MAP{$_} ||= 'label-default' } @ALL;

1;

__END__

=encoding utf8

=head1 NAME

OpenCloset::Constants::Status

=head1 EXPORT_OK

=over

=item $RENTABLE

=item $RENTAL

=item $RENTALESS

=item $RESERVATION

=item $CLEANING

=item $REPAIR

=item $LOST

=item $DISCARD

=item $RETURNED

=item $PARTIAL_RETURNED

=item $RETURNING

=item $NOT_VISITED

=item $VISITED

=item $RESERVATED

=item $RETURN_RESERVATED

=item $MEASUREMENT

=item $SELECT

=item $BOX

=item $PAYMENT

=item $FITTING_ROOM1

=item $FITTING_ROOM2

=item $FITTING_ROOM3

=item $FITTING_ROOM4

=item $FITTING_ROOM5

=item $FITTING_ROOM6

=item $FITTING_ROOM7

=item $FITTING_ROOM8

=item $FITTING_ROOM9

=item $FITTING_ROOM10

=item $FITTING_ROOM11

=item $FITTING_ROOM12

=item $FITTING_ROOM13

=item $FITTING_ROOM14

=item $FITTING_ROOM15

=item $FITTING_ROOM16

=item $FITTING_ROOM17

=item $FITTING_ROOM18

=item $FITTING_ROOM19

=item $FITTING_ROOM20

=item $NOT_RENTAL

=item $CANCEL_BOX

=item $PAYBACK

=item $NO_SIZE

=item $BOXED

=item $RECYCLE_1

=item $RECYCLE_2

=item $UNRECYCLE

=item $CHOOSE_CLOTHES

=item $PAYMENT_DONE

=item $DEPOSIT_CONFIRMED

=item $WAITING_SHIPPED

=item $SHIPPED

=item $DELIVERED

=item $RETURN_REQUESTED

=back

=head1 COPYRIGHT

The MIT License (MIT)

Copyright (c) 2016 열린옷장

=cut
